test/e2e-go/features/transactions/asset_test.go

```diff
@@ -143,15 +143,31 @@ func TestAssetValidRounds(t *testing.T) {
 	firstValid = 0
 	lastValid = 0
 
-	tx, err = client.FillUnsignedTxTemplate(account0, firstValid, lastValid, fee, tx)
+	params, err = client.SuggestedParams()
 	a.NoError(err)
+	lastRoundBefore := params.LastRound
 
+	tx, err = client.FillUnsignedTxTemplate(account0, firstValid, lastValid, fee, tx)
+	a.NoError(err)
 	// zeros are special cases
-	// first valid must be set to last round + 1 and never be zero
-	// last valid must be set to first + MaxTxnLife - 1
+	// first valid never should be zero
 	a.NotEqual(basics.Round(0), tx.FirstValid)
-	a.Equal(basics.Round(params.LastRound+1), tx.FirstValid)
-	a.Equal(basics.Round(params.LastRound+cparams.MaxTxnLife+1), tx.LastValid)
+
+	params, err = client.SuggestedParams()
+	a.NoError(err)
+	lastRoundAfter := params.LastRound
+
+	lastRound := lastRoundBefore
+	if lastRoundBefore != lastRoundAfter {
+		// round changed, not sure when, ask +-1
```

Comment:
This would be unlikely, but it might be possible that the ledger advanced _two_ rounds in between the calls, and thus neither would be correct. Maybe you could get rid of the if-else clauses and check that `tx.FirstValid` is between `lastRoundBefore` and `lastRoundAfter`, inclusive, and then check that `tx.LastValid` is correctly offset?

Reply:
I thought about this and this seems to be more relaxed condition than the current one. I can change it if you think this is ok

Reply:
done